src/lib/polygon.js

```diff
@@ -0,0 +1,240 @@
+/**
+* Copyright 2012-2015, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+var dot = require('./matrix').dot;
+
+/**
+ * Turn an array of [x, y] pairs into a polygon object
+ * that can test if points are inside it
+ *
+ * @param ptsIn Array of [x, y] pairs
+ *
+ * @returns polygon Object {xmin, xmax, ymin, ymax, pts, contains}
+ *      (x|y)(min|max) are the bounding rect of the polygon
+ *      pts is the original array, with the first pair repeated at the end
+ *      contains is a function: (pt, omitFirstEdge)
+ *          pt is the [x, y] pair to test
+ *          omitFirstEdge truthy means points exactly on the first edge don't
+ *              count. This is for use adding one polygon to another so we
+ *              don't double-count the edge where they meet.
+ *          returns boolean: is pt inside the polygon (including on its edges)
+ */
```

Comment:
oh right... that's where it started, when this was the only export :see_no_evil: